src/transformStateWithClones.js

```diff
@@ -8,6 +8,42 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const obj = {};
+  const arr = [];
+
+  for (const key in state) {
+    obj[key] = state[key];
+  }
+
+  for (let i = 0; i < actions.length; i++) {
+    const array = Object.values(actions[i]);
+    const data = array[1];
```

Comment:
looks like hardcode(((
```suggestion
  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;
```
you can use [Destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) here

Reply:
and switch by `type`